workspaces/adventure-pack/goodies/typescript/Math.gcd/index.ts

```diff
@@ -0,0 +1,12 @@
+declare global {
+  interface Math {
+    gcd(a: number, b: number): number;
+  }
+}
+
+Math.gcd = function (a: number, b: number): number {
+  while (a !== 0) {
+    [a, b] = [b % a, a];
```

Comment:
While I do like the destructuring swap, one downside of it is that I do think it has to create an array for the right hand side of the assignment. So I tend to avoid it in library code, where repeated array creation is an avoidable cost.